Settings.StyleCop

```diff
@@ -117,6 +163,61 @@
             <BooleanProperty Name="Enabled">False</BooleanProperty>
           </RuleSettings>
         </Rule>
+        <Rule Name="StaticElementsMustAppearBeforeInstanceElements">
+          <!-- it's OK to have static members after instance members -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+      </Rules>
+      <AnalyzerSettings />
+    </Analyzer>
+    <Analyzer AnalyzerId="StyleCop.CSharp.LayoutRules">
+      <Rules>
+        <Rule Name="ClosingCurlyBracketMustBeFollowedByBlankLine">
+          <!-- it's OK to not have a blank line after '}' -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+        <Rule Name="ElementsMustBeSeparatedByBlankLine">
+          <!-- it's OK to not have a blank line between adjacent members.
+               think about the 'Visit*' methods in Ast visitor class. -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+        <Rule Name="ElementMustNotBeOnSingleLine">
+          <!-- it's OK to have simple member on a single line.
+               think about the 'Visit*' methods in Ast visitor class. -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+        <Rule Name="StatementMustNotBeOnSingleLine">
+          <!-- it's OK to have a statement on a single line. -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+        <Rule Name="AllAccessorsMustBeSingleLineOrMultiLine">
+          <!-- it's OK to have the getter be on a single line and the
+               setter be on multiple lines, or vice versa -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+      </Rules>
+      <AnalyzerSettings />
+    </Analyzer>
+    <Analyzer AnalyzerId="StyleCop.CSharp.SpacingRules">
+      <Rules>
+        <Rule Name="MemberAccessSymbolsMustBeSpacedCorrectly">
+          <!-- it's OK to place a member name after '.' on the next line -->
```

Comment:
(!) It looks very bad. I guess we better fix this.

Reply:
In general, I'd agree doing this isn't the best idea. However, there are some constructs we use in the formatting subsystems when defining format views where we _do_ routinely use this kind of construct.

For example:

https://github.com/PowerShell/PowerShell/blob/f7eecba0d0d9e52526560ca11d9cb3f817311465/src/System.Management.Automation/FormatAndOutput/DefaultFormatters/Diagnostics_Format_ps1xml.cs#L26

And I'd argue in that kind of construct, as long as it's carefully dealt with in a consistent manner... this is fine to allow. I'dexpect that any particularly bad examples will be caught in code review anyway.

Reply:
Actually, I agree. I will submit a PR.

Reply:
@vexx32 the example you have above doesn't violate this rule. The following does:
```
TableControl.Create().
    AddHeader(Alignment.Left, label: "Timestamp", width: 25).
    AddHeader(Alignment.Left, label: "CounterSamples", width: 100).
    EndTable()); 
```

Reply:
Yeah, that makes sense. Thanks for the clarification! 😄

Reply:
I removed the change to this rule from this PR. The violations are fixed in #8517